proxy.go

```diff
@@ -73,7 +73,7 @@ func Join(local, remote net.Conn, log logging.Logger) {
 
 	transfer := func(side string, dst, src net.Conn) {
 		log.Debug("proxing %s -> %s", src.RemoteAddr(), dst.RemoteAddr())
-
+		defer dst.Close()
```

Comment:
While I agree the transfer function does not look ok to me, I don't think the proposed PR fixes it. I believe we should CloseRead on the dst instead, however I'd need put more thought into this.